.github/workflows/build-plugin-zip.yml

```diff
@@ -48,38 +141,62 @@ jobs:
           name: gutenberg-plugin
           path: ./gutenberg.zip
 
+      - name: Build release notes draft
+        if: ${{ needs.bump-version.outputs.new_version }}
+        env:
+          VERSION: ${{ needs.bump-version.outputs.new_version }}
+        run: |
+          IFS='.' read -r -a VERSION_ARRAY <<< "${VERSION}"
+          MILESTONE="Gutenberg ${VERSION_ARRAY[0]}.${VERSION_ARRAY[1]}"
+          npm run changelog -- --milestone="$MILESTONE" --unreleased > release-notes.txt
+          sed -ie '1,6d' release-notes.txt
+          if [[ ${{ needs.bump-version.outputs.new_version }} != *"rc"* ]]; then
+            # Include previous RCs' release notes, if any
+            CHANGELOG_REGEX="=\s[0-9]+\.[0-9]+\.[0-9]+(-rc\.[0-9]+)?\s="
+            RC_REGEX="=\s${VERSION}(-rc\.[0-9]+)?\s="
+            awk "/${RC_REGEX}/ {found=1;print;next} /${CHANGELOG_REGEX}/ {found=0} found" changelog.txt >> release-notes.txt
+          fi
+
+      - name: Upload release notes artifact
+        if: ${{ needs.bump-version.outputs.new_version }}
+        uses: actions/upload-artifact@e448a9b857ee2131e752b06002bf0e093c65e571 # v2.2.2
+        with:
+          name: release-notes
+          path: ./release-notes.txt
+
   create-release:
     name: Create Release Draft and Attach Asset
-    needs: build
+    needs: [bump-version, build]
```

Comment:
Seems like this could be just "build"  right?

Reply:
I think that would cause it to run for other triggers -- e.g. on a PR (since the `build` job is run in those cases, whereas the `bump-version` job isn't). Coupling it to `bump-version` basically makes sure it's only run when the workflow is triggered manually.